src/Form/DonationRequestType.php

```diff
@@ -92,6 +112,11 @@ public function createInitialDonationRequest(FormEvent $formEvent): void
         if (!$formEvent->getData()) {
             $user = $this->tokenStorage->getToken()->getUser();
 
+            // The user come from the registration process
+            if ($uuid = $this->membershipRegistrationProcess->getAdherentUuid()) {
+                $user = $this->entityManager->getRepository(Adherent::class)->findByUuid($uuid);
```

Comment:
You should inject the repository in your construct and use it here instead of using entityManager to get the repository.